projects/lexical_shared/lib/lexical/document/position.ex

```diff
@@ -70,6 +70,33 @@ defmodule Lexical.Document.Position do
         }
     end
   end
+
+  @doc """
+  Compares two positions.
+
+  Returns `:gt`, `:lt`, or `:eq` depending on the location of the first
+  position relative to the second.
+  """
+  @spec compare(t | {line, character}, t | {line, character}) :: :lt | :eq | :gt
+  def compare(%__MODULE__{} = pos1, %__MODULE__{} = pos2) do
+    compare({pos1.line, pos1.character}, {pos2.line, pos2.character})
+  end
+
+  def compare(%__MODULE__{} = pos1, pos2) do
+    compare({pos1.line, pos1.character}, pos2)
+  end
+
+  def compare(pos1, %__MODULE__{} = pos2) do
+    compare(pos1, {pos2.line, pos2.character})
+  end
+
+  def compare({_, _} = first, {_, _} = second) do
```

Comment:
this feels a little dangerous, should we constrain line and character to be integers?

Reply:
Yeah, you're right, especially considering it's ultimately doing term comparison. I'll add guards.